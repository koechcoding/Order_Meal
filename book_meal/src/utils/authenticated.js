export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}