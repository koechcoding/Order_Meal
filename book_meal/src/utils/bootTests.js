import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
