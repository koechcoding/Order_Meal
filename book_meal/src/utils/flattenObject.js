export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}