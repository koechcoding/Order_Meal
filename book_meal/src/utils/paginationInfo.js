export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}