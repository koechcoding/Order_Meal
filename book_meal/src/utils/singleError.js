const singleError = (errorPayload)=>{
    if(errorPayload && errorPayload.data){
        const response = errorPayload.data;
        //alternative to Object.keys...
        for(let errorKey in response.errors){
            return response.error[errorKey]
        }
        if(response.message){
            return response.message;
        }
    }
    return 'Unknown error. Please try again!';
}
export default singleError;