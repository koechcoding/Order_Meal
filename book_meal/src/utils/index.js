import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}