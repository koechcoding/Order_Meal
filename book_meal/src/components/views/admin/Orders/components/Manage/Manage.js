import React from 'react';
import PropTypes from 'prop-types';
import axios from 'src/axios';
import Modal from 'src/components/common/Modal';
import { Alert, Button } from 'reactstrap';
import { ToggleState } from 'src/constants';
import { singleError } from 'src/utils';


class Manage extends React.Component {

    state = {}

    onOpened = () => {
        this.setState({
            ...this.state,
            error: null,
        });
    }

    onSetToggleState = (status) => {
        let { order } = this.props;
        console.log(order);
        let newOrder = {
            user_id: order.user_id,
            menu_item_id: order.menu_item_id,
            status,
        }
        this.props.setLoading(true);
        axios.put(`/orders/${order.id}`, newOrder).then(() => {
            this.props.onChange();
            this.props.toggle();
            this.props.setLoading(false);
        }).catch(({ response }) => {
            this.setState({
                ...this.state,
                error: response,
            });
            this.props.setLoading(false);
        });
    }

    render() {
        const { error } = this.state;
        const { order = {} } = this.props;
        const body = (
            <div>
                {error &&
                    <Alert className="text-center text-small" color="danger">
                        { singleError(error) }
                    </Alert>
                }
                <p className="text-center">
                    Accept or reject this order(ID: <b>{order.id}</b>)
                </p>
            </div>
        );

        const footer = (
            <div className="m-auto">
                <Button 
                    onClick={() => this.onSetToggleState(ToggleState.REJECTED)} 
                    className="mr-2" 
                    color="danger">
                    Reject
                </Button>
                <Button 
                    onClick={() => this.onSetToggleState(ToggleState.ACCEPTED)}  
                    color="primary">
                    Accept
                </Button>
            </div>
        );

        const { isOpen, toggle } = this.props;
        return (
            <Modal 
                title="Manage Order Status" 
                body={body} 
                footer={footer} 
                isOpen={isOpen}
                toggle={toggle}
                onOpened={this.onOpened}
            />
        );
    }
}

Manage.propTypes = {
    setLoading: PropTypes.func.isRequired,
    isOpen: PropTypes.bool.isRequired,
    onChange: PropTypes.func.isRequired,
    order: PropTypes.object.isRequired
}

export default Manage;