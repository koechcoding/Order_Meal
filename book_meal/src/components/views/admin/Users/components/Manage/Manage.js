import React from 'reacxt';
import axios from 'src/axios';
import Modal from 'src/components/common/Modal';
import { Alert, Button } from 'reactstrap';
import { Role } from 'src/constants';
import { authenticated, singleError } from 'src/utils';

class ManageModal extends React.Component {

    state = {}

    onOpened =()=>{
        this.setState({
            ...this.state,
            error: null,
        });
    }

    onSetRole=(role)=>{
        const { user } = this.props;
        let newUser = {
            ...user,
            role,
        }
        const authUser = authenticated();
        this.props.setLoading(true);
        axios.put('users/${user.id', newUser).then(()=>{

            if(authUser.id === user.id){
                localStorage.clear();
                this.props.setLoading(false);
                setTimeout(()=>{
                    this.props.history.push('/login');
                    this.props.toggle();
                }, 1000);
            } else{
                this.props.toggle();
                this.props.setLoading(false);
                this.props.onChange();
            }
        }).catch(({ response }) =>{
            this.setState({
                ...this.state,
                error: response,
            });
            this.props.setLoading(false);
        });
    }
}