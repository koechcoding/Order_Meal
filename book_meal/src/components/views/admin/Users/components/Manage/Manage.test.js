import React from 'react';
import { shallow }  from 'enzyme';
import 'src/utils/bootTests'; 
import Manage from './Manage';
import Adapter from 'enzyme-adapter-react-16';

const props = {
    order: {},
    isOpen: false,
    toggle: () => {},
    onChange: () => {},
    setLoading: () => {},
};
describe('<Manage />', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Manage {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('resets on opened', () => {
        const wrapper = shallow(<Manage {...props}/>);
        wrapper.setState({...wrapper.state(), error: 'error'});
        wrapper.instance().onOpened();
        expect(wrapper.state('error')).toBe(null);
    });
});