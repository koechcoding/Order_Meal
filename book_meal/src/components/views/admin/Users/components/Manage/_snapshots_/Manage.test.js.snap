
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Manage /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ManageModal
    isOpen={false}
    onChange={[Function]}
    order={Object {}}
    setLoading={[Function]}
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "body": <div>
        
</div>,
      "footer": <div
        className="m-auto"
>
        
</div>,
      "isOpen": false,
      "onOpened": [Function],
      "title": "Manage User",
      "toggle": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "body": <div>
          
</div>,
        "footer": <div
          className="m-auto"
>
          
</div>,
        "isOpen": false,
        "onOpened": [Function],
        "title": "Manage User",
        "toggle": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;